src/query/vacuum.c

```diff
@@ -7482,42 +7379,18 @@ vacuum_log_redoundo_vacuum_record (THREAD_ENTRY * thread_p, PAGE_PTR page_p, PGS
   addr.pgptr = page_p;
   addr.vfid = NULL;
 
-  num_redo_crumbs = 0;
   if (reusable)
     {
       addr.offset |= VACUUM_LOG_VACUUM_HEAP_REUSABLE;
     }
-  else if (next_version_oid != NULL && !OID_ISNULL (next_version_oid))
-    {
-      buffer_p = PTR_ALIGN (buffer, MAX_ALIGNMENT);
-      ptr = buffer_p;
-      ptr = PTR_ALIGN (ptr, OR_OID_SIZE);
-      memcpy (ptr, next_version_oid, OR_OID_SIZE);
-
-      ptr += OR_OID_SIZE;
-      packed_size = OR_OID_SIZE;
-
-      if (partition_oid != NULL && !OID_ISNULL (partition_oid))
-	{
-	  /* Pack partition OIDs */
-	  memcpy (ptr, partition_oid, OR_OID_SIZE);
-	  ptr += OR_OID_SIZE;
-	  packed_size += OR_OID_SIZE;
-	}
-
-      redo_crumbs[0].length = packed_size;
-      redo_crumbs[0].data = buffer_p;
-      num_redo_crumbs = 1;
-    }
 
   undo_crumbs[0].length = sizeof (undo_recdes->type);
   undo_crumbs[0].data = (char *) &undo_recdes->type;
   undo_crumbs[1].length = undo_recdes->length;
   undo_crumbs[1].data = undo_recdes->data;
   num_undo_crumbs = 2;
 
-  log_append_undoredo_crumbs (thread_p, RVVAC_HEAP_RECORD_VACUUM, &addr, num_undo_crumbs, num_redo_crumbs, undo_crumbs,
-			      (num_redo_crumbs == 0) ? NULL : redo_crumbs);
+  log_append_undoredo_crumbs (thread_p, RVVAC_HEAP_RECORD_VACUUM, &addr, num_undo_crumbs, 0, undo_crumbs, NULL);
```

Comment:
It would be great if you can add a comment to describe the reason why we need to write undoredo log even with 0 redo_crumb.